src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -352,20 +349,19 @@ def _push_content(self) -> str:
                 elif self.format == 'full':
                     project_short_name = self._write_full(text_buffer)
                     if project_short_name:
-                        self._create_name_object(object_key, project_short_name)
+                        multipart_upload.file_name = self._create_name(project_short_name)
```

Comment:
Another bad smell: setting this attribute so late also indicates that tagging the object doesn't really have anything to do with the MP upload itself.